ens-improvement-proposals/ensip-17-evmchain-reverse-resolution.md

```diff
@@ -0,0 +1,109 @@
+* Dapps will then resolve this name via ENS on L1 to check if the forward resolution matches. This forward resolution can be on L1, or the user can set up a CCIP-read resolver and records for each network and put those addresses wherever they want (such as on another L2)
+* Once matched, the dapp can then also resolve any text records on the primary ENS name, such as avatar.
+* Discovery of the reverse registrar on each chain will be done by looking up the `addr()` of [coinTypeAsHex].reverse.
+* coinType in all instances will be the hex representation to reduce the length of the name
+
+### Deployment and discovery of EVM Reverse registrars
+
+When a new EVM reverse registrar is deployed, it will need to be setup by the owner of the `reverse` node, to setup a subdomain [coinTypeAsHex].reverse. It must then be setup with an Offchain resolver that has an onchain L1 address record that return the contract address of the reverse registrar for that chain. The Offchain resolver will also support wildcard of all the address subdomains with the format [address].[coinTypeAsHex].reverse. Additionally there must be a new EVM gateway setup to handle the CCIP-read revert errors that will go to the corresponding network to gather the chain-specific reverse record and verify this data on L1.
+
+### Resolving Primary ENS Names by a dapp
+
+1) If a dapp has a connected user it SHOULD first check the chainId of the user.
+2) It MUST then construct the cointype using ENSIP11: `coinType = 0x80000000 | chainId`, which must be converted to the hexadecimal representation.
+3) If the chainId is not 1, it SHOULD then construct the ENS node representing the reverse node on that network using the coinType and the user's connected `address`: `node = namehash([address].[coinTypeAsHex].reverse)`
+4) Call the registry with the namehash to retrieve the resolver `resolver = registry.resolver(node)`
+5) Call name on the resolver `name = resolver.name(node)`
+6) If a string is found, skip to step 12
+7) If `name` is an empty string, it SHOULD check the default primary name for all EVM chains by constructing its node as follows: `node = namehash([address].default.reverse)`.
+8) Call the registry with the namehash to retrieve the resolver `resolver = registry.resolver(node)`
+9) Call name on the resolver `name = resolver.name(node)`
+10) If a string is found, skip to step 12
+11) If `name` is an empty string, the dapp MUST assume that a primary name does not exist and instead show the address instead. The name resolution process ends here.
+12) If the dapp finds an ENS name, it MUST first check the forward resolution. This can be done by constructing the ENS node for the primary name found: `primaryName = namehash(name)`
+13) It MUST then call the registry to retrieve the resolver `registry.resolver(primaryName)`
```

Comment:
This implies not using ENSIP-11 wildcard resolution. Instead, this should just specify that it's to use the resolution process documented there.

Reply:
I'm guessing you meant ENSIP-10 here. Will add a reference to wildcard here